lib/http/printer.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+module HTTP
+  # Printer for {HTTP::Request} and {HTTP::Response}.
+  # Expects `print_headline` method to be implemented.
+  #
+  # @example Usage
+  #
+  #   class MyHttpRequest
+  #     include HTTP::Printer
+  #
+  #     def print_headline
+  #       "MyHttpRequest headline"
+  #     end
+  #   end
+  module Printer
+    # Returns human-readable representation including headers.
+    #
+    # @return [String]
+    def inspect
+      "#<#{pretty_print(:skip_headers => false)}>"
+    end
+
+    # Returns a printable representation of headers.
+    # @option [Hash] options
+    #   @param [Boolean] skip_headers
+    #   @param [Boolean] skip_body
+    #   @param [String] separator
+    # @return [String]
+    def pretty_print(skip_headers: true, skip_body: true, separator: ", ")
+      StringIO.open do |io|
+        io << print_headline
+        io << "#{separator}#{headers.pretty_print(separator)}" unless skip_headers
+        io << "#{separator}#{body}" unless skip_body || skip_headers
+        io.string
+      end
+    end
+
+    private
+
+    def print_headline
```

Comment:
`HTTP::Request#headline` does not return the entire URI. But for log/inspect purposes should be valid return the entire URI. 

For example:

```ruby
HTTP.get('http://prodis.blog.br', params: { s: 'pog' })
# Request healine is:
# "GET /?s=pog HTTP/1.1"

# For log/inspect should be:
# "GET http://prodis.blog.br/?s=pog HTTP/1.1"
```

We could modify the original `HTTP::Request#headline` to return the entire URI, but I don't know if this modification affects performance in the stream or socket to make requests. 

I did it in my tests and worked fine. I don't want to be intrusive and to cause side effects.